src/kbc_modules/eaa_kbc/mod.rs

```diff
@@ -165,23 +263,37 @@ impl EAAKbc {
                 anyhow!(format!("Something wrong when transmit, error code: {}", e))
             })?;
 
-        // At present, the returned packets of the verdictd server is small,
-        // and the size of these packets does not exceed 4096 bytes (we have tested),
-        // them will not be divided into multiple packets (if this happen, an error will be reported).
-        let mut buffer = [0u8; 4096];
-        let len_recv = self
-            .tls_handle
-            .as_ref()
-            .ok_or(anyhow!("Missing TLS handle"))?
-            .receive(&mut buffer)
-            .map_err(|e| {
-                error!("Transmit {} failed", error_info);
-                anyhow!(format!("Something wrong when revieve, error code: {}", e))
-            })?;
+        let mut recv_buffer = vec![0u8; rats_tls::MAX_FRAG_LENGTH];
+        let mut recv_res = vec![];
+        let mut len_recv = 0;
+
+        while len_recv < recv_size {
+            let len_single = self
+                .tls_handle
+                .as_ref()
+                .ok_or(anyhow!("Missing TLS handle"))?
+                .receive(&mut recv_buffer)
+                .map_err(|e| {
+                    error!("Transmit {} failed", error_info);
+                    anyhow!(format!("Something wrong when revieve, error code: {}", e))
+                })?;
 
-        let recv_string: String = String::from_utf8(buffer[..len_recv].to_vec())?;
+            recv_res.append(&mut recv_buffer[..len_single].to_vec());
```

Comment:
Is `receive` going to return something different when you call it multiple times? Otherwise it seems like you are just going to be appending the same thing to the buffer each time the loop runs.

Reply:
The receive here will return the data one by one in the form of slices in order. Because the maximum size of the data transmitted by TLS at one time is limited, each receive in the loop here will return different contents of the data.